Scripts/cours.js

```diff
@@ -56,47 +67,74 @@ function deleteAll(displayOnly = true){
 
     for(let i=size-1; i>1; i--){
         class_tab.deleteRow(i);
-    }
+    } 
+    showButton();
 }
 
 function deleteClass(id){
+    console.log(id);
+    if(classes == 0)
+        classes = rawTab;
  const idClass = classes.findIndex(c => c.id === id);
 
  if(idClass !== -1){
     classes.splice(idClass, 1);
     deleteAll();
-    getClasses();
+    displayClasses();
  }else{
     alert("Not found");
  }
 }
 
-function postClass(){
+function createClass(){
     const title = document.getElementById("titleInput").value;
     const period = document.getElementById("periodInput").value;
     const nbHour = document.getElementById("nbHourInput").value;
     const teacher = document.getElementById("teacherInput").value;
     const description = document.getElementById("descriptionInput").value;
 
-    if(!title || !period || !teacher || !description || !Number.isInteger(+nbHour)){
+    // Ctrl input
+    // Number.isInteger(+nbHour) => le + permet de retirer les ""
+    if(!title || !period || !teacher || !description || !nbHour)
```

Comment:
you didn't check if the nbHour is an integer

Reply:
Done, now!